spec/03-plugins/02-legacy_propagation_parameter_warning_spec.lua

```diff
@@ -0,0 +1,124 @@
+local cjson      = require "cjson"
+local helpers    = require "spec.helpers"
+
+
+for _, strategy in helpers.each_strategy() do
+  describe("legacy propagation parameters [#" .. strategy .. "]", function()
+    local db
+    local admin_client
+
+    lazy_setup(function()
+      -- Create a service to make sure that our database is initialized properly.
+      local bp
+      bp, db = helpers.get_db_utils(strategy, {
+        "services",
+      })
+
+      db:truncate()
+
+      bp.services:insert{
+        protocol = "http",
+        host     = helpers.mock_upstream_host,
+        port     = helpers.mock_upstream_port,
+      }
+      assert(helpers.start_kong({
+        database = strategy,
+        nginx_conf = "spec/fixtures/custom_nginx.template",
+      }))
+      admin_client = helpers.admin_client()
+    end)
+
+    lazy_teardown(function()
+      if admin_client then
+        admin_client:close()
+      end
+      helpers.stop_kong()
+    end)
+
+    before_each(function()
+      helpers.clean_logfile()
+    end)
+
+    local plugin_id
+
+    after_each(function()
+      if plugin_id then
+        local res = admin_client:delete("/plugins/" .. plugin_id)
+        assert.res_status(204, res)
+      end
+    end)
+
+    local plugins = {
+      ["zipkin"] = {
+        http_endpoint = "http://example.com/",
+      },
+      ["opentelemetry"] = {
+        endpoint = "http://example.com/",
+      },
+    }
+
+    for plugin, base_config in pairs(plugins) do
+
+      local function create_plugin(parameter, value)
+        local config = table.clone(base_config)
+        if parameter then
+          config[parameter] = value
+        end
+
+        local res = admin_client:post(
+          "/plugins",
+          {
+            headers = {
+              ["Content-Type"] = "application/json"
+            },
+            body = cjson.encode({
+              name = plugin,
+              config = config
+            })
+          }
+        )
+        local body = cjson.decode(assert.res_status(201, res))
+        plugin_id = body.id
+      end
+
+      local log_wait_time = 0.01
```

Comment:
Is this large enough?

Reply:
got it from [here](https://github.com/Kong/kong/blob/ed0b96de49da824d76122cbaf7fc15060405afe5/spec/03-plugins/01-legacy_queue_parameter_warning_spec.lua#L85), it was enough for that test and all the runs I tried on this one.